cpp/src/groupby/hash/single_pass_functors.cuh

```diff
@@ -15,14 +15,213 @@
  */
 #pragma once
 
-#include <cudf/detail/aggregation/aggregation.hpp>
+#include "helpers.cuh"
+
 #include <cudf/detail/aggregation/device_aggregators.cuh>
-#include <cudf/groupby.hpp>
-#include <cudf/utilities/bit.hpp>
+#include <cudf/detail/utilities/assert.cuh>
+#include <cudf/detail/utilities/device_atomics.cuh>
+#include <cudf/utilities/traits.cuh>
 
-#include <cuco/static_set_ref.cuh>
+#include <cuda/std/cstddef>
 
 namespace cudf::groupby::detail::hash {
+
+// TODO: TO BE REMOVED
+template <typename T, cudf::aggregation::Kind k>
+__device__ constexpr bool is_supported()
+{
+  return cudf::is_fixed_width<T>() and
+         ((k == cudf::aggregation::SUM) or (k == cudf::aggregation::MIN) or
+          (k == cudf::aggregation::MAX) or (k == cudf::aggregation::COUNT_VALID) or
+          (k == cudf::aggregation::COUNT_ALL) or (k == cudf::aggregation::ARGMAX) or
+          (k == cudf::aggregation::ARGMIN) or (k == cudf::aggregation::SUM_OF_SQUARES) or
+          (k == cudf::aggregation::STD) or (k == cudf::aggregation::VARIANCE) or
+          (k == cudf::aggregation::PRODUCT) and cudf::detail::is_product_supported<T>());
```

Comment:
I know this is silly but if you move `SUM_OF_SQUARES` next to `SUM` you may get a nice symmetry to this layout
```suggestion
         ((k == cudf::aggregation::SUM) or (k == cudf::aggregation::SUM_OF_SQUARES) or
          (k == cudf::aggregation::MIN) or (k == cudf::aggregation::MAX) or
          (k == cudf::aggregation::COUNT_VALID) or (k == cudf::aggregation::COUNT_ALL) or
          (k == cudf::aggregation::ARGMIN) or (k == cudf::aggregation::ARGMAX) or
          (k == cudf::aggregation::STD) or (k == cudf::aggregation::VARIANCE) or
          (k == cudf::aggregation::PRODUCT) and cudf::detail::is_product_supported<T>());
```
(Sorry, I could not help myself).